rclcpp/src/rclcpp/time.cpp

```diff
@@ -141,84 +87,62 @@ Time::operator!=(const rclcpp::Time & rhs) const
 bool
 Time::operator<(const rclcpp::Time & rhs) const
 {
-  if (rcl_time_.clock_type != rhs.rcl_time_.clock_type) {
-    throw std::runtime_error("can't compare times with different time sources");
-  }
-
+  // will throw if not comparable
+  __comparable(rcl_time_.clock_type, rhs.rcl_time_.clock_type);
   return rcl_time_.nanoseconds < rhs.rcl_time_.nanoseconds;
 }
 
 bool
 Time::operator<=(const rclcpp::Time & rhs) const
 {
-  if (rcl_time_.clock_type != rhs.rcl_time_.clock_type) {
-    throw std::runtime_error("can't compare times with different time sources");
-  }
-
+  // will throw if not comparable
+  __comparable(rcl_time_.clock_type, rhs.rcl_time_.clock_type);
   return rcl_time_.nanoseconds <= rhs.rcl_time_.nanoseconds;
 }
 
 bool
 Time::operator>=(const rclcpp::Time & rhs) const
 {
-  if (rcl_time_.clock_type != rhs.rcl_time_.clock_type) {
-    throw std::runtime_error("can't compare times with different time sources");
-  }
-
+  // will throw if not comparable
+  __comparable(rcl_time_.clock_type, rhs.rcl_time_.clock_type);
   return rcl_time_.nanoseconds >= rhs.rcl_time_.nanoseconds;
 }
 
 bool
 Time::operator>(const rclcpp::Time & rhs) const
 {
-  if (rcl_time_.clock_type != rhs.rcl_time_.clock_type) {
-    throw std::runtime_error("can't compare times with different time sources");
-  }
-
+  // will throw if not comparable
+  __comparable(rcl_time_.clock_type, rhs.rcl_time_.clock_type);
   return rcl_time_.nanoseconds > rhs.rcl_time_.nanoseconds;
 }
 
 Time
 Time::operator+(const rclcpp::Duration & rhs) const
 {
-  if (rclcpp::add_will_overflow(rhs.nanoseconds(), this->nanoseconds())) {
-    throw std::overflow_error("addition leads to int64_t overflow");
-  }
-  if (rclcpp::add_will_underflow(rhs.nanoseconds(), this->nanoseconds())) {
-    throw std::underflow_error("addition leads to int64_t underflow");
-  }
-  return Time(this->nanoseconds() + rhs.nanoseconds(), this->get_clock_type());
+  rcl_time_point_value_t res = 0;
+  // will throw if addition over/underflows
+  check_add_overflow(rcl_time_.nanoseconds, rhs.nanoseconds(), &res);
```

Comment:
somehow I can't help it but the function names are a bit confusing for me. So it really does two things, it checks for overflow and performs the addition (same for sub, mult). 
Can we not just rename this to `add`? Given from the comment and signature of the function it's clear what's happening on overflow.